### ISSUE TITLE
No User Guide chapter for Lens Distortion plugin

### ISSUE THREAD

**Issue by gzotti** [Stellarium/stellarium#3525]
### Expected Behaviour

All plugins are described in the User Guide

### Actual Behaviour

The new Lens Distortion Estimator Plugin is not described in the User Guide. I assume it is of interest to astrophotographers or just photographers who need a way to find lens parameters for other applications (which? Name a few examples. ). 

Please describe the plugin and what it can be used for (practical examples) in the User Guide. 1-2 images, 2-3 pages. 

On a different idea, once a match has been accomplished, can this be used for an entry in nebulae\...\textures.json? It may require exporting the undistorted image. Or to find angle/coordinate parameters when creating a SkyImage? Or to link a piece of sky art to three stars? (To use as constellation art in skycultures.)

**Comment by 10110111** [comment]
> can this be used for an entry in nebulae...\textures.json? It may require exporting the undistorted image.

Yes, may be a good use. Do you have an example of a distorted input image?

> Or to find angle/coordinate parameters when creating a SkyImage?

Do you mean the parameters for image placement? If so, yes, these can also be added to the generator.

> Or to link a piece of sky art to three stars? (To use as constellation art in skycultures.)

Not sure why you need this plugin for such a task. Doesn't it simply require you knowing pixel position of the stars?

As for distortion, the constellation art images seem to be quite fuzzy regarding the position of the stars. Is there any example where you'd need a precise placement of an image with respect to the stars?

**Comment by gzotti** [comment]
> > can this be used for an entry in nebulae...\textures.json? It may require exporting the undistorted image.
> 
> Yes, may be a good use. Do you have an example of a distorted input image?

You can use any self-made DSO image. I can PM you some. Probably lens distortion is less of an issue to telescope images, but image placement in J2000 coordinates would be: find image corner coordinates. 

 
> > Or to find angle/coordinate parameters when creating a SkyImage?
> 
> Do you mean the parameters for image placement? If so, yes, these can also be added to the generator.

Yes, there are arguments for either position&angle, or four corner coordinates. When showing some large image insert in a planetarium show, the four corners would be fine to avoid large distortion on the dome. In this use case, output should be alt-azimuthal coordinates, but J2000 coordinates should also be available for "real" sky insertions. 
 
> > Or to link a piece of sky art to three stars? (To use as constellation art in skycultures.)
> 
> Not sure why you need this plugin for such a task. Doesn't it simply require you knowing pixel position of the stars?

Indeed, currently users load the image into some image editor and read pixel coordinates of stars visible in the figures. This plugin finally could allow interactive placement. This would also allow moving the image where it fits best, then selecting 3 stars and finding the pixel coordinates in the image.

> As for distortion, the constellation art images seem to be quite fuzzy regarding the position of the stars. Is there any example where you'd need a precise placement of an image with respect to the stars?

Yes, the 3-star method just makes a planar stretch through whatever stars were selected. Matching an old copperplate map sometimes is problematic. When the original map projection is not stereographic or perspective (not even sure what's best) with triangle balance point as image centre, distortions or mismatches are hardly avoidable. Now, if the figure could be distorted and re-exported, it may fit better. Again, I will PM you some scans.

**Comment by gzotti** [comment]
Another application: Find lens parameters (f, fov) when sensor size is known.

**Comment by 10110111** [comment]
I must admit that, while the computation seems in principle correct, the result appears to have an error of several pixels due to the stars being hard to pick because of being blurred+noisy even at the best focus. Additionally, increasing exposure time to reduce noise leads to motion blur due to diurnal motion of the Earth, which again complicates picking.

In particular, for my ~65°×51° Galaxy S8 back camera the error is ~0.03° in the best case, which makes simple stitching of two overlapping shots noticeably imperfect. I don't think I'll be able to improve much on this before the release.

**Comment by gzotti** [comment]
To stitch I prefer to use Hugin and let it match adjacent shots using multiple stars identified in both images. Sure, you must limit exposure time or track the camera to avoid star streaks. The biggest problem to match wide-angle shots with Stellarium's sky may be refraction under real-world conditions. To find lens parameters for short-focus lenses, probably only zenith shots would work.

**Comment by 10110111** [comment]
Yeah, my original aim was to avoid Hugin's guesswork, because it resulted in broken horizon (which is very visible on seascapes). The workflow would be:

1. Shoot the sky near zenith
2. Find distortion parameters as well as the real FoV—all from a single photo
3. Do whatever shots of the landscape you wanted and use the parameters found in step 2 to rectify their geometry.

But unfortunately, imperfect focus and noise result in the difficulties I described above. I think they are in principle solvable, but this needs quite a bit of work (like e.g. picking the stars and radii they should be within according to the corresponding blobs in the image, and then making a fit to get each star into its circle rather than try to get it as close as possible to a point).

**Comment by 10110111** [comment]
Due to the problems mentioned above (and some additional ones I've noticed) I consider this plugin experimental, and thus not worth putting into the User Guide yet. There are some docs in the _About_ dialog, so it's not undocumented. But until I find a way to resolve the problems I don't think lots of people need to know about it.

@alex-w If you're going to mention it in the release notes, please include the word "experimental" there.

**Comment by alex-w** [comment]
@10110111 any news?

**Comment by 10110111** [comment]
While this plugin is still experimental I don't want to draw too much attention to it. It does have the docs in the description, let it be like this for now.

**Comment by gzotti** [comment]
Presumably this will also be pushed further?

**Comment by 10110111** [comment]
Yes.
